pkg/littlefs/fs/littlefs_fs.c

```diff
@@ -85,12 +85,21 @@ static int _dev_write(const struct lfs_config *c, lfs_block_t block,
     DEBUG("lfs_write: c=%p, block=%" PRIu32 ", off=%" PRIu32 ", buf=%p, size=%" PRIu32 "\n",
           (void *)c, block, off, buffer, size);
 
-    int ret = mtd_write(mtd, buffer, ((fs->base_addr + block) * c->block_size) + off, size);
-    if (ret >= 0) {
-        return 0;
+    const uint8_t *buf = buffer;
+    uint32_t addr = ((fs->base_addr + block) * c->block_size) + off;
+    for (size_t i = 0; i < size / c->prog_size; i++) {
```

Comment:
Can you move the division out of the loop, or get rid of it completely?

Reply:
The function now uses a memory pointer

Reply:
thx!